### ISSUE TITLE
The 'Search With Google' action in macOS' context menu doesn't respect the 'Default web browser' setting

### ISSUE THREAD

**Issue by n5a5** [eclipse-platform/eclipse.platform.swt#1215]
**Describe the bug**
On macOS, the 'Search With Google' action in the context menu opens Google search in Safari even if another browser is set as default.

**To Reproduce**
1. On macOS, set a non-safari browser as default (I used Brave)
1. In Eclipse IDE for RCP and RAP devs 2024-03, click Eclipse -> About Eclipse -> Installation Details -> Configuration.
1. Select any text, invoke the context menu, and click 'Search With Google'.
1. The selected text is searched in Safari.

**Expected behavior**
The text should be searched in the default browser.

**Environment:**
1. Select the platform(s) on which the behavior is seen:
- - [ ] All OS
- - [ ] Windows
- - [ ] Linux
- - [x] macOS 

3. Additional OS info (e.g. OS version, Linux Desktop, etc)

macOS 14.4.1.

4. JRE/JDK version
17.0.6, 17.0.10+7

**Workaround (or) Additional context**
To be fair, native macOS applications such as TextEdit and Mail do not respect this setting either. However, all the third-party apps I tried (Slack, IntelliJ IDEA, Notion) invoke the default browser, and it seems like the correct behavior.

Originally reported here: https://github.com/dbeaver/dbeaver/issues/26611.

**Comment by Phillipus** [comment]
I doubt this can be changed. The SWT Browser on Mac is basically Safari, and selecting "Search With Google" on selected text in Safari itself will open another Safari tab or window.

**Comment by JTMosaic** [comment]
Highlighting a word in Slack on MacOS and selecting "Search with Google..." opens in the default browser. So, this should be possible, no?

**Comment by Phillipus** [comment]
> Highlighting a word in Slack on MacOS and selecting "Search with Google..." opens in the default browser. So, this should be possible, no?

It seems that any _native_ Mac app such as Safari, Messages, Contacts, Notes, and so on, that has the "Search with Google" options will open it in Safari. The SWT Browser is a native Mac component (Safari).

**Comment by Phillipus** [comment]
> Turning off the built-in "Search with Google" service allows Eclipse to use the default browser.

@JTMosaic How did you do that? Disabling it in the Services dialog has no effect for me (Sonoma).

**Comment by JTMosaic** [comment]
I just noticed something. There are **two** different context-based searches in DBeaver (based on Eclipse)

There is a one in free-text, such as a script that reads: 
![image](https://github.com/eclipse-platform/eclipse.platform.swt/assets/9417720/86668836-c626-4d01-ae5a-e9aba147649a)

and one that appears in the results pane that reads:
![image](https://github.com/eclipse-platform/eclipse.platform.swt/assets/9417720/2d3fbe9f-c3a0-486f-bf40-768c4271d7a5)

The latter appears to use SWT and only opens in Safari while the former, as the tooltip says, opens in the default browser.

This would seem to indicate that it is possible for Eclipse to use the default browser but it isn't implemented consistently?

**Comment by JTMosaic** [comment]
So you're saying that vanilla Eclipse doesn't have the "Search selected text with Google" option? Then this may be a DBeaver issue and not an Eclipse issue, per se. Assuming the results grid isn't a SWT component.

**Comment by n5a5** [comment]
> I would think that DBeaver added that option as a custom menu item.
> So you're saying that vanilla Eclipse doesn't have the "Search selected text with Google" option?

That's correct. It's a custom menu item added by DBeaver.

Thank you for clarifications @Phillipus. Now we have a couple of questions left: a) How does Slack respect the default browser setting, and b) Do we want the Eclipse Platform to follow Slack's route?

In any case, I find this a pretty minor thing, and I would rather see SWT developers spending time on something more substantial. What say you @JTMosaic?

**Comment by Phillipus** [comment]
> Thank you for clarifications @Phillipus. Now we have a couple of questions left: a) How does Slack respect the default browser setting

I don't use Slack so I don't know. Is the the context menu there the same as Mac's native Service menu.

>  and b) Do we want the Eclipse Platform to follow Slack's route?

For the reasons above this will not be possible. The "Search with Google" menu item is provided by Safari as a "service". I doubt anyone will spend any time on this. ;-)

**Comment by JTMosaic** [comment]
@n5a5 I agree. I see this as a DBeaver issue more than an Eclipse issue but I think the discussion was helpful for all involved.
